pkg/kn/commands/service_get_flags.go

```diff
@@ -0,0 +1,182 @@
+
+// AllowedFormats is the list of formats in which data can be displayed
+func (f *ServiceGetFlags) AllowedFormats() []string {
+	formats := f.GenericPrintFlags.AllowedFormats()
+	formats = append(formats, f.HumanReadableFlags.AllowedFormats()...)
+	return formats
+}
+
+// ToPrinter attempts to find a composed set of ServiceGetFlags suitable for
+// returning a printer based on current flag values.
+func (f *ServiceGetFlags) ToPrinter() (printers.ResourcePrinter, error) {
+	// if there are flags specified for generic printing
+	if f.GenericPrintFlags.OutputFlagSpecified() {
+		p, err := f.GenericPrintFlags.ToPrinter()
+		if err != nil {
+			return nil, err
+		}
+		return p, nil
+	}
+	// if no flags specified, use the table printing
+	p, err := f.HumanReadableFlags.ToPrinter()
+	if err != nil {
+		return nil, err
+	}
+	return p, nil
+}
+
+// AddFlags receives a *cobra.Command reference and binds
+// flags related to humanreadable and template printing.
+func (f *ServiceGetFlags) AddFlags(cmd *cobra.Command) {
+	f.GenericPrintFlags.AddFlags(cmd)
+	f.HumanReadableFlags.AddFlags(cmd)
+}
+
+// NewGetPrintFlags returns flags associated with humanreadable,
+// template, and "name" printing, with default values set.
+func NewServiceGetFlags() *ServiceGetFlags {
+	return &ServiceGetFlags{
+		GenericPrintFlags:  genericclioptions.NewPrintFlags(""),
+		HumanReadableFlags: NewHumanPrintFlags(),
+	}
+}
+
+// ServiceGetHandlers adds print handlers for service get command
+func ServiceGetHandlers(h hprinters.PrintHandler) {
+	kServiceColumnDefinitions := []metav1beta1.TableColumnDefinition{
+		{Name: "Name", Type: "string", Description: "Name of the knative service."},
+		{Name: "Domain", Type: "string", Description: "Domain name of the knative service."},
+		//{Name: "LastCreatedRevision", Type: "string", Description: "Name of last revision created."},
+		//{Name: "LastReadyRevision", Type: "string", Description: "Name of last ready revision."},
+		{Name: "Generation", Type: "integer", Description: "Sequence number of 'Generation' of the service that was last processed by the controller."},
+		{Name: "Age", Type: "string", Description: "Age of the service."},
+		{Name: "Conditions", Type: "string", Description: "Conditions describing statuses of service components."},
+		{Name: "Ready", Type: "string", Description: "Ready condition status of the service."},
+		{Name: "Reason", Type: "string", Description: "Reason for non-ready condition of the service."},
+	}
+	h.TableHandler(kServiceColumnDefinitions, printKService)
+	h.TableHandler(kServiceColumnDefinitions, printKServiceList)
+}
+
+// conditionsValue returns the True conditions count among total conditions
+func conditionsValue(conditions duckv1alpha1.Conditions) string {
+	var ok int
+	for _, condition := range conditions {
+		if condition.Status == "True" {
+			ok++
+		}
+	}
+	return fmt.Sprintf("%d OK / %d", ok, len(conditions))
+}
+
+// readyCondition returns status of resource's Ready type condition
```

Comment:
Following few functions should be grouped together in some core lib to re-used in other resource listing. I plan to regroup them post v0.1 part of some refactoring.

Reply:
+1